app/services/hmpps_api/assessment_api.rb

```diff
@@ -0,0 +1,20 @@
+module HmppsApi
+  class AssessmentApi
+    def self.client
+      host = Rails.configuration.assessment_api_host
+      HmppsApi::Client.new(host)
+    end
+
+    def self.get_latest_oasys_date(offender_no)
+      safe_offender_no = URI.encode_www_form_component(offender_no)
+      route = "/offenders/nomisId/#{safe_offender_no}/assessments/summary?assessmentType=LAYER_3&assessmentStatus=COMPLETE"
+
+      all_assessments = self.client.get(route)
+
+      if all_assessments.is_a?(Array) && all_assessments.present?
+        # return the latest assessment date
+        all_assessments.map { |assessment| assessment.fetch('completed').to_date }.max
+      end
```

Comment:
Given my previous comment about the "offender not found" scenario actually returning a 404 rather than 200 – it means that it should be possible to make this a little nicer.

Hint: so long as we get a `200 OK` response, we should always have an array. It's only when a 404 error code is returned that we receive a hash.